x-pack/plugins/apm/server/lib/search_strategies/correlations/query_correlation.ts

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { estypes } from '@elastic/elasticsearch';
+
+import type { ElasticsearchClient } from 'src/core/server';
+
+import { TRANSACTION_DURATION } from '../../../../common/elasticsearch_fieldnames';
+import type { SearchServiceParams } from '../../../../common/search_strategies/correlations/types';
+
+import { getQueryWithParams } from './get_query_with_params';
+
+export interface HistogramItem {
+  key: number;
+  doc_count: number;
+}
+
+interface ResponseHitSource {
+  [s: string]: unknown;
+}
+interface ResponseHit {
+  _source: ResponseHitSource;
+}
+
+interface BucketCorrelation {
+  buckets_path: string;
+  function: {
+    count_correlation: {
+      indicator: {
+        expectations: number[];
+        doc_count: number;
+      };
+    };
+  };
+}
+
+export const getTransactionDurationCorrelationRequest = (
+  params: SearchServiceParams,
+  expectations: number[],
+  ranges: estypes.AggregationsAggregationRange[],
+  fractions: number[],
+  totalDocCount: number,
+  fieldName?: string,
+  fieldValue?: string
+): estypes.SearchRequest => {
+  const query = getQueryWithParams({ params, fieldName, fieldValue });
+
+  const bucketCorrelation: BucketCorrelation = {
+    buckets_path: 'latency_ranges>_count',
+    function: {
+      count_correlation: {
+        indicator: {
+          expectations,
+          doc_count: totalDocCount,
+        },
+      },
+    },
+  };
+
+  const body = {
+    query,
+    size: 0,
+    aggs: {
+      latency_ranges: {
+        range: {
+          field: TRANSACTION_DURATION,
+          ranges,
+        },
+      },
+      // Pearson correlation value
+      transaction_duration_correlation: {
+        bucket_correlation: bucketCorrelation,
+      } as estypes.AggregationsAggregationContainer,
+      // KS test p value = ks_test.less
+      ks_test: {
+        bucket_count_ks_test: {
+          fractions,
```

Comment:
I think here, you need to prepend a `0` to the fractions, otherwise you will get an error saying that the sizes don't match up.